### PR TITLE
Bittrex v3 endpoints

### PR DESCRIPTION
**Pull request by npomfret**
* added v3 api endpoints + and added supporting code in sign method
 * fixed timestamp in deposits
 * replaced fetchClosedOrders with v3 api
 * fixed fee bug in parseTransaction

### CODE REVIEW

File: js/bittrex.js [hunk 1]
```diff
@@ -768,6 +825,80 @@ module.exports = class bittrex extends Exchange {
...
+            'id': this.safeString (order, 'id'),
+            'timestamp': timestamp,
+            'datetime': this.iso8601 (timestamp),
+            'lastTradeTimestamp': lastTradeTimestamp,
+            'symbol': symbol,
+            'type': type.toLowerCase (),
+            'side': direction.toLocaleLowerCase (),
+            'price': limit,
+            'cost': proceeds,
+            'average': proceeds / fillQuantity,
```
Comment (kroitor) [inline js/bittrex.js]: A possible division by zero here.

**Comment by rdbr** [comment]
There is a problem with the new v3 `fetchClosedOrders`, calling it with correct symbol `FLO/BTC`:
```
bittrex: malformed response: {\"code\":\"MARKET_NAME_REVERSED\",\"detail\":\"The provided market symbol appears to be reversed. Please retry with the market symbol provided in data.NewMarketSymbol.\",\"data\":{\"newMarketSymbol\":\"FLO-BTC\"}}
```

When using `'fetchClosedOrdersMethod': 'fetch_closed_orders_v2'` everything works, so maybe the symbol format changed in v3?

**Comment by rdbr** [comment]
Another problem, when using `cancelOrder`:
```
Cannot read property 'split' of undefined
```

cancelOrder uses `parseOrder` while the API only returns:
```
{
  "success": true,
  "message": "''",
  "result": {
    "uuid": "614c34e4-8d71-11e3-94b5-425861b86ab6"
  }
}
```

This kicks in the `parseOrderV3`.

**Comment by kroitor** [comment]
@rdbr thx, will add fixes and workarounds shortly!

**Comment by kroitor** [comment]
@npomfret check this out: https://github.com/ccxt/ccxt/blob/master/js/bittrex.js#L1148
I think I will cut out bittrex v3 into a separate bittrex3 file, because otherwise all methods become a real mess, fetchMarkets, fetchOrders, sign(), API definitions, etc...

**Comment by npomfret** [comment]
@kroitor that's a shame.  I don't fully understand the problem defined in the comment, but I can see that the class is more complicated.

The major issue I have with multiple classes for a single exchange __as a user of CCXT__ is that invariably neither implementation has the _best_ set of methods.  I have to use both bitfinex and bitfinex2 in my application, and juggling the exchange ids, synchronising the nonce etc is a massive pain.  And, as a user it's never obvious which implementation to pick.

In this particular case:

 * could we delete the old method and just stick with the new one? 
 * if you do go down the 2 classes route, can you inherit?  I normally dislike inheritance, but from a users point of view, the alternative (having two implementations) is very difficult to work with
 * could we have a couple of utility methods in the class that we use to do the symbol mapping that are _aware_ of which api the data comes from / goes to?
 * could we re-write the entire thing in v3?

I'd be happy to help with the last one (except I'm away next week).

**Comment by kroitor** [comment]
> I don't fully understand the problem defined in the comment, but I can see that the class is more complicated.

The problem is that Bittrex initially had market ids _"reversed"_ as `USDT-BTC` where BTC is the base and USDT is the quote, so, the corresponding unified symbol would be `BTC/USDT`. However, in v3 they've re-reversed those ids, so now they're straight: `BTC-USDT`. And all parsers and all fetch methods rely on market ids which are different depending on the API version. Each market has only one id though. Either we add an additional Bittrex-specific field for reversed market ids to avoid unnecessary string-concatenation and all the mess with market ids in the code of the parsers and everywhere around the file, or we need to invent something different, if we want to keep it a single class. I'll think a bit more on this... No worries, I'm not going to break it.

> could we re-write the entire thing in v3?

Yes, ultimately, this is the best way of solving this.

**Comment by rdbr** [comment]
The v3 is still in beta with a warning https://bittrex.github.io/api/v3
`Warning: The v3-Beta or other pre-release API versions are not for production use.`

I think it makes sense to create a v3 version in addition to the production-ready version. v3 functionality should not be used by live applications yet.

One other issue, overriding the hostname doesn't work for v3:
```
'hostname': 'international.bittrex.com',
```

**Comment by kroitor** [comment]
> One other issue, overriding the hostname doesn't work for v3:

@rdbr it should work with the most recent version of CCXT, what's your version?

**Comment by kroitor** [comment]
@npomfret 

> I think it makes sense to create a v3 version in addition to the production-ready version. v3 functionality should not be used by live applications yet.

I agree with @rdbr on this, so, if Bittrex says we should not rely on it, then, maybe we should develop the entire v3 implementation and just offer bittrex3.js for some time in parallel to the stable bittrex.js ?

**Comment by rdbr** [comment]
Overriding the hostname works, but when applied to:
`'v3': 'https://api.{hostname}/v3',` its a wrong url

International is needed when using `fetchOHLCV` from the v2 api to get the candles from geo-restricted markets.

**Comment by npomfret** [comment]
For me the hostname isn't the problem.  When you have two implementations they invariably have different features.  Maybe because CCXT hasn't _finished_ the new implementation, or that the old and new underlying APIs have different features.  In my code I call bitfinex v1 occasionally because their new API in some places isn't as good as the old one.  It's difficult for me to manage. 

As a user of ccxt when you see two implementations it presents an immediate problem.

Perhaps we could have 2 _abstract_ classes, and a top level Bittrex class that we can effectively configure method by method? 

The reason I added v3 is that I has way better support for historical data.  It's a big improvement .

**Pull request merged**